pkg/querier/queryrange/limits.go

```diff
@@ -416,23 +426,75 @@ func (sl *seriesLimiter) Do(ctx context.Context, req queryrangebase.Request) (qu
 		return res, nil
 	}
 	sl.rw.Lock()
+
 	var hash uint64
+	newResult := make([]queryrangebase.SampleStream, 0, len(promResponse.Response.Data.Result))
```

Comment:
I don't like this, it's going to double the memory needed. I'm going to change this on Tuesday to reuse the slice in the response.